InformationScripting/src/queries/AstQuery.cpp

```diff
@@ -69,16 +73,23 @@ Optional<TupleSet> AstQuery::executeLinear(TupleSet input)
 
 void AstQuery::registerDefaultQueries()
 {
-	registerQuery("classes", &AstQuery::genericQuery, "Class");
-	registerQuery("methods", &AstQuery::genericQuery, "Method");
-	registerQuery("toClass", &AstQuery::toParentType, "Class");
+	registerQuery("classes", &AstQuery::genericQuery, {}, "Class");
+	registerQuery("methods", &AstQuery::genericQuery, {}, "Method");
+	registerQuery("toClass", &AstQuery::toParentType, {}, "Class");
 	registerQuery("bases", &AstQuery::baseClassesQuery);
 	registerQuery("callgraph", &AstQuery::callGraph);
-	registerQuery("ast", &AstQuery::genericQuery);
-	registerQuery("toParent", &AstQuery::toParentType);
-	registerQuery("uses", &AstQuery::usesQuery);
-	registerQuery("type", &AstQuery::typeFilter);
-	registerQuery("attribute", &AstQuery::attribute);
+	registerQuery("ast", &AstQuery::genericQuery,
+		{{Arguments::RequireOneOf(), "ast requires either name or type argument",
```

Comment:
ideally you should not have to provide a string, but it should be automatically deduced.